sdk/go/auto/cmd.go

```diff
@@ -61,6 +252,71 @@ func runPulumiCommandSync(
 	return stdout.String(), stderr.String(), code, err
 }
 
+// The version of the current Pulumi CLI installation.
+func (p pulumiCommand) Version() semver.Version {
+	return p.version
+}
+
+type pulumiCommandOptions struct {
+	version          semver.Version
+	root             string
+	skipVersionCheck bool
+}
+
+// PulumiCommandOption is used configure a PulumiCommand at initialization and
+// installation time.
+type PulumiCommandOption interface {
+	applyPulumiCommandOption(*pulumiCommandOptions)
+}
+
+type pulumiCommandOption func(*pulumiCommandOptions)
+
+func (o pulumiCommandOption) applyPulumiCommandOption(opts *pulumiCommandOptions) {
+	o(opts)
+}
+
+// PulumiCommandVersion is the version to install or validate.
+func PulumiCommandVersion(version semver.Version) PulumiCommandOption {
+	return pulumiCommandOption(func(o *pulumiCommandOptions) {
+		o.version = version
+	})
+}
+
+// PulumiCommandRoot sets the directory where the CLI should be installed to,
+// or from where the CLI should be retrieved.
+func PulumiCommandRoot(root string) PulumiCommandOption {
```

Comment:
Picking up the thread here https://github.com/pulumi/pulumi/pull/15049#discussion_r1461976161

I renamed it the options to `PulumiCommandRoot` and `PulumiCommandVersion`. I left `skipVersionCheck` as is, since it's purely internal.